spec/services/checksum_validator_spec.rb

```diff
@@ -123,4 +123,98 @@
       end
     end
   end
+
+  context '#validate_signature_catalog' do
+    let(:druid) { 'bj102hs9687' }
+    let(:cv) { described_class.new(druid, endpoint_name) }
+
+    it 'instantiates storage_object from druid and druid_path' do
+      expect(Moab::StorageObject).to receive(:new).with(cv.druid, a_string_matching(object_dir)).and_call_original
+      cv.validate_signature_catalog
+    end
+
+    it 'calls validate_signature_catalog_entry for each signatureCatalog entry' do
+      sce01 = instance_double(Moab::SignatureCatalogEntry)
+      sce02 = instance_double(Moab::SignatureCatalogEntry)
+      sce03 = instance_double(Moab::SignatureCatalogEntry)
+      sce04 = instance_double(Moab::SignatureCatalogEntry)
+      sce05 = instance_double(Moab::SignatureCatalogEntry)
+      sce06 = instance_double(Moab::SignatureCatalogEntry)
+      sce07 = instance_double(Moab::SignatureCatalogEntry)
+      sce08 = instance_double(Moab::SignatureCatalogEntry)
+      sce09 = instance_double(Moab::SignatureCatalogEntry)
+      sce10 = instance_double(Moab::SignatureCatalogEntry)
+      sce11 = instance_double(Moab::SignatureCatalogEntry)
+      entry_list = [sce01, sce02, sce03, sce04, sce05, sce06, sce07, sce08, sce09, sce10, sce11]
+      moab_storage_object = instance_double(Moab::StorageObject)
+      allow(cv).to receive(:moab_storage_object).and_return(moab_storage_object)
+      allow(cv).to receive(:latest_signature_catalog_entries).and_return(entry_list)
+      entry_list.each do |entry|
+        expect(cv).to receive(:validate_signature_catalog_entry).with(entry)
+      end
+      cv.validate_signature_catalog
+    end
+
+    it 'calls AuditResults.report_results' do
+      results = instance_double(AuditResults, add_result: nil, :actual_version= => nil, :check_name= => nil)
+      allow(AuditResults).to receive(:new).and_return(results)
+      expect(results).to receive(:report_results)
+      cv.validate_signature_catalog
+    end
+
+    context 'modify a file signature in signatureCatalog.xml' do
```

Comment:
` 'file checksums in singatureCatalog.xml do not match' ` just to keep it constant w/ manifestInv?